### PR TITLE
server side rendering for custom pagination and added search parameter

### PR DESCRIPTION
**Pull request by onkartibe**
server side rendering for go to page for custom pagination and added search parameter in the state while navigating in the pagination

### CODE REVIEW

File: src/components/Table.js [hunk 1]
```diff
@@ -169,17 +171,22 @@ export class Table extends Component {
...
+                return goToPage({newPage: pageNumber, state: currentState, shouldCall})
+            });
+            return
+        }
 
-        this.setState(currentState => {
-            return goToPage({newPage: pageNumber, state: currentState, shouldCall})
-        })
+        const { sort, search } = this.state;
+        if (pageNumber > 1) this.props.updateData({ page: pageNumber, sort, search: search });
```
Comment (JoeJoeDev) [inline src/components/Table.js]: do we need to check that pageNumber is greater than 1 or should we check that the pageNumber > 0 and <= totalPages within the goToPage function?

**Review by JoeJoeDev — CHANGES_REQUESTED** [review]
do we need to check that pageNumber is greater than 1 or should we check that the pageNumber > 0 and <= totalPages within the goToPage function?

File: src/components/Table.js (hunk 1 above)
Comment (onkartibe) [reply src/components/Table.js]: good point.. will make the changes.

Comment (onkartibe) [reply src/components/Table.js:189]: good point

**Review by JoeJoeDev — APPROVED** [review]

**Pull request merged**